segmentation/utils/spatialanalysis_utils_test.py

```diff
@@ -1,12 +1,310 @@
+    # marker_thresholds = pd.read_csv(
+    #     "/Users/jaiveersingh/Downloads/SpatialEnrichment/markerThresholds.csv")
+    # dist_matrix = np.asarray(pd.read_csv(
+    #     "/Users/jaiveersingh/Documents/MATLAB/distancesMat5.csv", header=None))
+    # closenum, closenumRand, z, muhat, sigmahat, p, h, adj_p, markertitles = \
+    #     spatialorgmarkerexpression_utils.calculate_channel_spatial_enrichment(
+    #         dist_matrix, marker_thresholds, cell_array)
+    # real_closenum = np.asarray(pd.read_csv(
+    #     "/Users/jaiveersingh/Documents/MATLAB/SpatialAnalysis/closeNum.csv"))
+    # assert np.array_equal(closenum, real_closenum)
+
+
+def test_closenumrand():
+    test_cellarray, test_distmat = make_test_closenum()
+
+    # generate random inputs to test shape
+    marker1_num = [random.randrange(0, 10) for i in range(20)]
+    marker2_num = [random.randrange(0, 5) for i in range(400)]
+    marker_num = 20
+
+    test_closenumrand = spatialorgmarkerexpression_utils.compute_close_cell_num_random(
+        marker1_num, marker2_num, test_cellarray, test_distmat, marker_num, dist_lim=100,
+        cell_label_idx=24, bootstrap_num=100)
+
+    assert test_closenumrand.shape == (20, 20, 100)
+
+
+def test_calculate_enrichment_stats():
+    # positive enrichment
+
+    # generate random closenum matrix
+    stats_cnp = np.zeros((20, 20))
+    stats_cnp[:, :] = 80
+
+    # generate random closenumrand matrix, ensuring significant positive enrichment
+    stats_cnrp = np.random.randint(1, 40, (20, 20, 100))
+
+    test_z, muhat, sigmahat, test_p, h, adj_p = spatialorgmarkerexpression_utils.calculate_enrichment_stats(
+        stats_cnp, stats_cnrp)
+
+    assert test_z[0, 0] > 0
+    assert test_p[0, 0, 0] < .05
+
+    # negative enrichment
+
+    # generate random closenum matrix
+    stats_cnn = np.zeros((20, 20))
+
+    # generate random closenumrand matrix, ensuring significant negative enrichment
+    stats_cnrn = np.random.randint(40, 80, (20, 20, 100))
+
+    test_z, muhat, sigmahat, test_p, h, adj_p = spatialorgmarkerexpression_utils.calculate_enrichment_stats(
+        stats_cnn, stats_cnrn)
+
+    assert test_z[0, 0] < 0
+    assert test_p[0, 0, 1] < .05
+
+    # no enrichment
+
+    # generate random closenum matrix
+    stats_cn = np.zeros((20, 20))
+    stats_cn[:, :] = 80
+
+    # generate random closenumrand matrix, ensuring no enrichment
+    stats_cnr = np.random.randint(78, 82, (20, 20, 100))
+
+    test_z, muhat, sigmahat, test_p, h, adj_p = spatialorgmarkerexpression_utils.calculate_enrichment_stats(
+        stats_cn, stats_cnr)
+
+    assert abs(test_z[0, 0]) < 1
+    assert test_p[0, 0, 1] > .05
+    assert test_p[0, 0, 0] > .05
+
+
+def test_spatial_analysis():
+    # test z and p values
+
+    marker_thresholds = make_threshold_mat()
+
+    # positive enrichment
+    cellarrayp = make_expression_matrix("positive")
+    distmatp = make_distance_matrix("positive")
+    values, stats = \
```

Comment:
This is good. I would then unpack the tuple so it's more intuitive what's being tested here

```
values, stats = function_output()
p_values, z_scores, _, _, _ = stats
assert (p_values[0, 0] == )

```